### PR TITLE
make gpu work

### PR DESCRIPTION
**Pull request by barni2000**
These are necessary changes for make a512 work on lavender.

Last two commits it not mergeable just example usage with lavender td4320 display variant.

Changes:
- Disable preemption for a512 (maybe a509 and 508 should be added also)
- Add zap-shader node for adreno_gpu
- TEMP add generated td4320 driver for lavender
- TEMP enable display and gpu on lavender
- remove zap_shader_region deletion from xiaomi-tulip dt

TODO:
- [ ] split lavender dt because of there are different display/touch variants
- [ ] enable/fix gpu opp
- [x] check preemption on a509 and a508

### CODE REVIEW

File: drivers/gpu/drm/msm/adreno/a5xx_gpu.c [hunk 1]
```diff
@@ -1770,7 +1770,7 @@ struct msm_gpu *a5xx_gpu_init(struct drm_device *dev)
 
 	nr_rings = 4;
 
-	if (config->info->revn == 510)
+	if (config->info->revn == 510 || config->info->revn == 512)
```
Comment (minlexx) [inline drivers/gpu/drm/msm/adreno/a5xx_gpu.c]: the same likely can be done for 508 & 509, probably (for devices based on sdm630 and sdm636) ?

File: drivers/gpu/drm/panel/msm8916-generated/panel-xiaomi-lavender-td4320.c [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+static int td4320_boeplus_prepare(struct drm_panel *panel)
+{
+	struct td4320_boeplus *ctx = to_td4320_boeplus(panel);
+	struct device *dev = &ctx->dsi->dev;
+	int ret;
+
+	if (ctx->prepared)
+		return 0;
+
+	td4320_boeplus_reset(ctx);
```
Comment (minlexx) [inline drivers/gpu/drm/panel/msm8916-generated/panel-xiaomi-lavender-td4320.c:103]: I remember you told it only works with reset commented out. It it still true?

**Comment by minlexx** [comment]
I'm ready to pick most of the commits from here to future 6.8 branch, except 2 last ( generated driver and temp one )

File: drivers/gpu/drm/msm/adreno/a5xx_gpu.c (hunk 1 above)
Comment (minlexx) [reply drivers/gpu/drm/msm/adreno/a5xx_gpu.c]: I do not think we have devices with sdm630/a508 to test, but @M0Rf30 I think has tulip with a509 ?

Comment (barni2000) [reply drivers/gpu/drm/msm/adreno/a5xx_gpu.c]: As far as i know a5xx preemption is broken globaly and there are parts what are not implemented in mesa so i could disable for every a5xx with remove the if.

File: drivers/gpu/drm/panel/msm8916-generated/panel-xiaomi-lavender-td4320.c (hunk 2 above)
Comment (barni2000) [reply drivers/gpu/drm/panel/msm8916-generated/panel-xiaomi-lavender-td4320.c:103]: No, the issue was i have used ```reset-gpios = <&tlmm 53 GPIO_ACTIVE_HIGH>;``` but it works fine with ```reset-gpios = <&tlmm 53 GPIO_ACTIVE_LOW>;```

**Comment by M0Rf30** [comment]
> I'm ready to pick most of the commits from here to future 6.8 branch, except 2 last ( generated driver and temp one )

Just FYK, also qcom_fq commits maybe should be removed in favour of the updated commit
https://gitlab.com/sdm845-mainline/linux/-/commit/b5f9aa03a205d74b7b4e2470cb92c258040849d5

**Pull request closed without merging**